File: 20. Valid Parentheses/20.py
class Solution:
    def isValid(self, s: str) -> bool:
        pairs = {'(': ')', '[': ']', '{': '}'}
        
        for index in range(0, len(s), 2):
            if pairs[s[index]] != s[index + 1]:
                return False
                
        return True

File: 20. Valid Parentheses/20.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {

    const pairs = {
        '(': ')',
        '[': ']',
        '{': '}'
    };
    
    for (let index = 0; index < s.length; index = index + 2) {
        if (pairs[s[index]] !== s[index+1]) {
            return false
        }

    }
    return true
};
